### PR TITLE
add feature importances

### PR DESCRIPTION
**Pull request by wxchan**
to solve #40

### CODE REVIEW

File: src/boosting/gbdt.cpp [hunk 1]
```diff
@@ -349,6 +353,18 @@ void GBDT::ModelsFromString(const std::string& model_str, int num_used_model) {
...
 }
 
+std::string GBDT::FeatureImportance(int last_iter) const {
+    size_t* feature_importances = new size_t[max_feature_idx_ + 1]{0};
+    for (int iter = 0; iter < last_iter; ++iter) {
+        for (int split_idx = 0; split_idx < models_.at(iter)->num_leaves() - 1; ++split_idx) {
+            ++feature_importances[models_.at(iter)->split_feature(split_idx)];
+        }
+    }
+    std::string ret = Common::ArrayToString(feature_importances, max_feature_idx_ + 1, ' ');
```
Comment (guolinke) [inline src/boosting/gbdt.cpp]: can we make this more readable? like feature_1: importance1 \n feature_2:importance2 \n ...

File: src/boosting/gbdt.cpp [hunk 2]
```diff
@@ -222,6 +224,8 @@ void GBDT::Train() {
       }
       fflush(output_model_file);
   }
+  fprintf(output_model_file, "feature importances=%s\n", FeatureImportance(models_.size()).c_str());
+  fflush(output_model_file);
```
Comment (guolinke) [inline src/boosting/gbdt.cpp]: I think it is better to put "feature importances=%s\n" into FeatureImportance function.

File: include/LightGBM/boosting.h [hunk 3]
```diff
@@ -80,6 +80,12 @@ class Boosting {
   virtual void ModelsFromString(const std::string& model_str, int num_used_model) = 0;
 
   /*!
+  * \brief Calculate feature importances
+  * \param last_iter Last tree use to calculate
+  */
+  virtual std::string FeatureImportance(int last_iter) const = 0;
```
Comment (guolinke) [inline include/LightGBM/boosting.h]: I think this is not need to expose as interface of boosting.h . just using as private member function in gbdt class

File: src/boosting/gbdt.cpp (hunk 1 above)
Comment (wxchan) [inline src/boosting/gbdt.cpp]: But we don't know feature names. It looks a little mess to me by using feature_1, feature_2 and so on. Is it possible you could change this part after merge?

**Pull request merged**